internal/kafka/test/integration/cluster_metrics_test.go

```diff
@@ -0,0 +1,94 @@
+package integration
+
+import (
+	"testing"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/constants"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/api/dbapi"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/kafkas/types"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/common"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/test/mocks/kasfleetshardsync"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/metrics"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/test/mocks"
+	. "github.com/onsi/gomega"
+)
+
+func TestClusterCapacityUsedMetric(t *testing.T) {
+	// create a mock ocm api server, keep all endpoints as defaults
+	// see the mocks package for more information on the configurable mock server
+	ocmServer := mocks.NewMockConfigurableServerBuilder().Build()
+	defer ocmServer.Close()
+
+	h, _, teardown := test.NewKafkaHelperWithHooks(t, ocmServer, nil)
+	defer teardown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	// setup pre-requisites to performing requests
+	db := test.TestServices.DBFactory.New()
+
+	// create a dummy cluster and assign a kafka to it
+	cluster := &api.Cluster{
+		ClusterID:             "some-cluster-id",
+		MultiAZ:               true,
+		Region:                "us-east-1",
+		CloudProvider:         "aws",
+		Status:                api.ClusterReady,
+		IdentityProviderID:    "some-id",
+		ClusterDNS:            "some-cluster-dns",
+		ProviderType:          api.ClusterProviderStandalone,
+		SupportedInstanceType: "standard",
+		ClientID:              "some-client-id",
+		ClientSecret:          "some-client-secret",
+	}
```

Comment:
Can we build this using the cluster? https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/mocks/clusters/clusters.go

Reply:
Thanks, I was looking at https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/mocks/clusters/clusters.go#L68 and I feel like I missing something. 
Since I need to control most of the things e.g clientIDs, identity provider id, I need to pass it a `modifyFn` which takes in the `*api.cluster` which'll have to built the same way I am doing now. So why go that extra step while I can just do like what I am doing which is the same as passing in the `modifyFn`?

Reply:
For me it should allow us to be consistent with a mock object for these tests rather than using different values for things like ids, etc. If we need to add new fields to a struct, we could also just add it to that mock without having to update each tests individually. 

Does it need to be specific values for those fields or can it be some standard default values? :thinking:

Reply:
The can be any values, I care for the values to be there so avoid for the secrets to be auto created. 

There is something about the build method which I do not understand i.e the modifyFn to take in something that it is building. If someone can pass in the `api.cluster`, then there is no really need to call the builder at all since they know how to build it.

Reply:
Ah just noticed those are missing in that one. The Cluster mock may be a bad example :thinking: I think the goal was to have a builder to something like this: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/mocks/kafkas/kafkas.go#L152

@ziccardi may be able to fill the gaps here? :)

Reply:
Thanks. The method was confusing as well as I was reading it wrong.
Having a signature like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/test/mocks/kafkas/kafkas.go#L152 would be best

Reply:
done